health-services/plan-service/src/main/java/digit/util/QueryUtil.java

```diff
@@ -121,6 +119,63 @@ public String preparePartialJsonStringFromFilterMap(Map<String, String> filterMa
         return partialJsonQueryString;
     }
 
+    /**
+     * This is an overloaded function, to handle facility multi select for plan search specifically.
+     * Prepares a partial JSON query string based on the provided filter map, key, and prepared statement list.
+     *
+     * @param filterMap         A map containing filter keys and their corresponding set of values.
+     * @param preparedStmtList  A list to which placeholder values for prepared statements are added.
+     * @param key               The key used to construct the partial query.
+     * @return                  A partial JSON query string or a query fragment with OR conditions.
+     *
+     * This method dynamically constructs a JSON query or query fragment based on the structure of the `key`:
+     * - For nested keys (keys containing dot separators), a nested JSON structure is constructed.
+     * - For the "facilityId" key, an OR condition is generated for all values in the set.
+     * - For other keys, a simple key-value JSON structure is returned.
+     *
+     * Notes:
+     * - When constructing the query for "facilityId", placeholders for each value are added to the `preparedStmtList`.
+     */
+    public String preparePartialJsonStringFromFilterMap(Map<String, Set<String>> filterMap, List<Object> preparedStmtList, String key) {
+        Map<String, Object> queryMap = new HashMap<>();
+        StringBuilder finalJsonQuery = new StringBuilder();
+        // Handle nested keys (dot separator)
+        if (key.contains(DOT_SEPARATOR)) {
```

Comment:
The implementationn is not generic here and will have to be taken up as part of tech debts since this was a last minute change.

Reply:
Yes.